ui/packages/catalog-ui-search/src/main/webapp/react-component/select-coordinates/select-coordinates.tsx

```diff
@@ -0,0 +1,237 @@
+ * to handle drawing the map markers and displays the measured distances.
+ */
+const coordHandler = (
+  context: ContextType,
+  closeParent: () => void,
+  selectCoordHandler: () => void,
+  mapModel: Backbone.Model
+) => {
+  return () => {
+    selectCoordHandler()
+
+    // displays an announcement with the current distance when the end state has been reached
+    if (mapModel.get('measurementState') === 'END') {
+      const distance = mapModel.get('currentDistance')
+      announcement.announce({
+        title: 'Distance between points:',
+        message: `${getDistanceText(distance)}`,
+        type: 'success',
+      })
+    }
+
+    context.closeAndRefocus()
+    closeParent()
+  }
+}
+
+/*
+ * Handler function for clearing all markers on the map. Calls clearRulerHandler() from props
+ * to handle removing the map markers from the map.
+ */
+const clearHandler = (
+  context: ContextType,
+  closeParent: () => void,
+  clearRulerHandler: () => void
+) => {
+  return () => {
+    clearRulerHandler()
+
+    announcement.announce({
+      title: 'Ruler markers cleared.',
+      message: '',
+      type: 'success',
+    })
+
+    context.closeAndRefocus()
+    closeParent()
+  }
+}
+
+/*
+ * Handler function for copying the measured distance string to the clipboard.
+ */
+const generateClipboardHandler = (
+  text: string,
+  context: ContextType,
+  closeParent: () => void
+) => {
+  return (e: React.MouseEvent) => {
+    const clipboardInstance = new Clipboard(e.target, {
+      text: () => {
+        return text
+      },
+    })
+    clipboardInstance.on('success', (e: any) => {
+      announcement.announce({
+        title: 'Copied to clipboard',
+        message: e.text,
+        type: 'success',
+      })
+    })
+    clipboardInstance.on('error', (e: any) => {
+      announcement.announce({
+        title: 'Press Ctrl+C to copy',
+        message: e.text,
+        type: 'info',
+      })
+    })
+    clipboardInstance.onClick(e)
+    clipboardInstance.destroy()
+    context.closeAndRefocus()
+    closeParent()
+  }
+}
+
+const render = (props: Props) => {
+  const { lat, lon } = props.coordinateValues
+  const { closeParent, selectCoordHandler, clearRulerHandler, mapModel } = props
+  const distanceText = getDistanceText(mapModel.get('currentDistance'))
+  const state = mapModel.get('measurementState')
+  let measurementSelectText = ''
+
+  // determines the text to display on the menu action based on the current measurement state
+  switch (state) {
+    case 'NONE':
+      measurementSelectText = 'Select start point'
+      break
+    case 'START':
+      measurementSelectText = 'Select end point'
+      break
+    case 'END':
+      measurementSelectText = 'Select new start point'
+      break
+    default:
+      break
+  }
+
+  return (
+    <CustomDropdown
+      content={(context: ContextType) => (
+        <NavigationBehavior>
+          <MenuAction
```

Comment:
I agree that we only really need "copy current distance" when we are in the "END" state. However, I'm concerned that if the menu does not have that option, users will not know that the "copy current distance" functionality exists, since previously when drawing the line, they did not see that as an option. We could try to figure out how to make the option unclickable though, if you think that would be an acceptable compromise.

Reply:
I agree that it should be an option in the menu, but be disabled/unclickable.

Reply:
Left the option in the menu, but it now displays `----` as the measurement value and is not copy-able.